### DISCUSSION TITLE
Widget configuration

### DISCUSSION THREAD

**Discussion by stefanolissa** [filamentphp/filament#6062] [Help] [answered]
Hi,

I created a widget (a chart widget) to display the current record's historic X1 value (think of it as a number from a sensor stored every hour). I have other values for this sensor, X2, X3, X4, X5, and for each of them I want to add a widget. But widgets are "added" (on getHeaderWidgets()) specifying the class name and not an instance that can be constructed or configured to display one of the Xs values.

The only solution that comes to my mind is to subclass my original widget and on each subclass deal with one of the Xs values: is it the correct way?  

Thank you!

**Comment by danharrin** [comment, marked as answer]
You probably need a custom widget where you can render multiple cards within it, instead of using our builtin widgets
